content/articles/getting-started-with-unit-testing-with-spring-boot/index.md

```diff
@@ -0,0 +1,131 @@
+#### Services
+This is the layer where all the business logic is implemented. It is tested to make sure that the business logic is correct.
+### How to test controllers
+Let us now look at how to test controllers in Spring Boot. To accomplish this, you will need to import some dependencies using the `Spring Initializer` and `IntelliJ IDEA.` `IntelliJ IDEA` lets us create a Spring Boot application using the Spring Initializr service. Open `file>new>project` and select `Spring Initializr` as shown below:
+
+![Spring Initializer](/engineering-education/getting-started-with-unit-testing-with-spring-boot/spring-Initializer.png)
+
+> Note: You can rename the Package name.
+
+Click next to proceed. Type *web* to search the required dependencies in the search bar. Select *Spring Web*, *Spring Reactive Web*, *Spring Web Services*, and click *finish* to download the initializr template:
+
+![Spring Initializer](/engineering-education/getting-started-with-unit-testing-with-spring-boot/spring-Initializer-Dependency.png)
+
+We've prepared the environment successfully. Now we need to test the Controller.
+In `/src/main/java` create a new package named `Controller`. Proceed to create a Java class named `HelloContoller.java` in  `/src/main/java/Controller`. Within that file, add the code below:
+```java
+package controller;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+@RestController
+public class HelloController {
+    @GetMapping("/hello")
+    public String hello(@RequestParam(name = "name", defaultValue = "Hello world") String name){
+        return String.format("Hello, %s", name );
+    }
+}
+```
+Let us look at the code snippet above:
+- `@GetMapping` is an annotation used to map the request to the method.
+- `@RequestParam` is an annotation used to map the request parameter to the method parameter.
+- `@RestController` is an annotation used to indicate that the class is a controller.
+
+Let us now generate the controller test.  In the `HelloController.java` file, right-click and select `Generate..>Test..` in the menu. Select the available method (which we want to test) under the member and click ok.
+
+![Spring Initializer](/engineering-education/getting-started-with-unit-testing-with-spring-boot/testing.png)
+
+A new file `/src/main/test/java/controller/HelloControllerTest.java` will be created. Modify the file to have the code below:
+```java
+package controller;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.*;
+class HelloControllerTest {
+    @Test
+    void hello() {
+        HelloController controller = new HelloController(); // instance of the controller
+        String response = controller.hello("Hello world"); // act
+        Assertions.assertEquals("Hello world", response); // assert
+    }
```

Comment:
The test fails:

![Capture](https://user-images.githubusercontent.com/63324129/160873407-937f9de2-ec3b-45c1-8cf5-df9ee27f7d69.PNG)